tempest/api/workloadmgr/base.py

```diff
@@ -3619,4 +3620,106 @@ def install_qemu(self, ssh):
+            rt_af, vm_details_bf, vm_details_af, test_type):
+        try:
+            if nt_bf == nt_af:
+                reporting.add_test_step(
+                    "Verify network details after network restore",
+                    tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify network details after network restore",
+                    tvaultconf.FAIL)
+                LOG.error(
+                    "Network details before and after restore: {0}, {1}".format(
+                        nt_bf, nt_af))
+
+            if sbnt_bf == sbnt_af:
+                reporting.add_test_step(
+                    "Verify subnet details after network restore",
+                    tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify subnet details after network restore",
+                    tvaultconf.FAIL)
+                LOG.error(
+                    "Subnet details before and after restore: {0}, {1}".format(
+                        sbnt_bf, sbnt_af))
+
+            if rt_bf == rt_af:
+                reporting.add_test_step(
+                    "Verify router details after network restore",
+                    tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify router details after network restore",
+                    tvaultconf.FAIL)
+                LOG.error(
+                    "Router details before and after restore: {0}, {1}".format(
+                        rt_bf, rt_af))
+
+            if test_type.lower() == 'cli':
+                if not vm_details_af:
+                    reporting.add_test_step("Instances not restored",
+                        tvaultconf.PASS)
+                else:
+                    reporting.add_test_step("Instances are restored",
+                        tvaultconf.FAIL)
+            else:
+                klist = sorted([*vm_details_bf])
+                vm_details_bf_sorted = {}
+                vm_details_af_sorted = {}
+
+                for vm in klist:
+                    netname = [*vm_details_bf[vm]['addresses']]
+                    for net in netname:
+                        for i in range((len(vm_details_bf[vm]['addresses'][net]))):
+                            vm_details_bf[vm]['addresses'][net][i]['OS-EXT-IPS-MAC:mac_addr'] = ''
+                            vm_details_af[vm]['addresses'][net][i]['OS-EXT-IPS-MAC:mac_addr'] = ''
+                    vm_details_bf[vm]['links'][1]['href'] = ''
+                    vm_details_af[vm]['links'][1]['href'] = ''
+                    if 'config_drive' in vm_details_af[vm]['metadata']:
+                        del vm_details_af[vm]['metadata']['config_drive']
+                    if 'ordered_interfaces' in vm_details_af[vm]['metadata']:
+                        del vm_details_af[vm]['metadata']['ordered_interfaces']
+                    vm_details_bf[vm]['links'] = ''
```

Comment:
You can reduce the below code from 3690 to 3707 using following code,

attribute_list = ['links', 'OS-EXT-SRV-ATTR:host', 'OS-EXT-SRV-ATTR:hypervisor_hostname', 'hostId', 'OS-EXT-SRV-ATTR:instance_name', 'updated','id', 'OS-SRV-USG:launched_at']

for index in attribute_list:
	vm_details_bf[vm][index] = ''
	vm_details_af[vm][index] = ''

Reply:
Done